pgxn/neon/walproposer.h

```diff
@@ -269,21 +270,30 @@ typedef struct HotStandbyFeedback
 
 typedef struct PageserverFeedback
 {
+	/* true if AppendResponse contains this feedback */
+	bool		present;
 	/* current size of the timeline on pageserver */
 	uint64		currentClusterSize;
 	/* standby_status_update fields that safekeeper received from pageserver */
 	XLogRecPtr	last_received_lsn;
 	XLogRecPtr	disk_consistent_lsn;
 	XLogRecPtr	remote_consistent_lsn;
 	TimestampTz replytime;
+	uint32		shard_number;
```

Comment:
Using a u32 here does no harm, but currently shard numbers are `u8`-sized (the `ShardNumber` type) elsewhere, so probably makes sense to match that.

Reply:
At the time of choosing u32 I was wondering if we're going to stay limited under 256 shards for a long time. Because changing the type here later can be painful, and all other types in this feedback are u64 anyway.

If we're sure that u8 is here to stay, I can change it to u8.

Reply:
If it's a binary persistent struct and you'd like to make it future proof, using u32 is fine -- we might indeed increase the shard count might higher in future.  If it's just used at runtime, then I'd use u8 and it'll just be a place we update as/when we use wider types in future.

Reply:
Not exactly a binary persistent struct, but we serialize and deserialize the binary bytelength (now it's 4):
```
Assert(value_len == sizeof(uint32));
ps_feedback->shard_number = pq_getmsgint(reply_message, sizeof(uint32));
psfeedback_log("%u", key, ps_feedback->shard_number);
```

This part makes it hard to change in the future, so yes, I'd want u32 to make it future proof.